tests/Doctrine/Tests/Annotations/DocParserTest.php

```diff
@@ -1255,6 +1255,21 @@ public function testSetValuesExeption()
         $this->createTestParser()->parse($docblock, 'some class');
     }
 
+    /**
+     * @expectedException \Doctrine\Annotations\AnnotationException
+     * @expectedExceptionMessage [Creation Error] The annotation @SomeAnnotationClassNameWithoutConstructorAndProperties declared on some class does not have a property
```

Comment:
This exception message does not correlate to any of the messages that this patch provides. Maybe you're testing another piece of code, but it seems to me that this test does not cover the added code.

Can you please add the test covering the implemented scenarios?